### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by JuanBohorquezPerficient**

### CODE REVIEW

File: cypress/page/index.ts [hunk 1]
```diff
@@ -7,3 +7,5 @@ export {ShippingStepPage} from "./shipping-step.page"
 export {PaymentStepPage} from "./payment-step.page"
 export {DressesListPage} from "./dresses-list.page"
 export {PersonalFormPage} from "./personal-form.page"
+export {Upload} from "./upload.page"
+
```
Comment (holgiosalos) [inline cypress/page/index.ts]: remove this line

File: package.json [hunk 2]
```diff
@@ -31,6 +31,9 @@
     "@typescript-eslint/eslint-plugin": "^5.30.5",
     "@typescript-eslint/parser": "^5.30.5",
     "cypress": "^10.3.0",
+
```
Comment (holgiosalos) [inline package.json]: remove this line

File: package.json [hunk 3]
```diff
@@ -31,6 +31,9 @@
...
+    "cypress-file-upload": "^5.0.8",
+
```
Comment (holgiosalos) [inline package.json]: remove this line

File: cypress/page/index.ts (hunk 1 above)
Comment (JuanBohorquezPerficient) [reply cypress/page/index.ts]: ok, Holmes I'm gonna do it... thx

File: cypress/page/upload.page.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    public uploadFile(targetFile: string): void{
+        cy.get(this.fileContainer).attachFile(targetFile);
+        cy.get(this.submitBtn).click({force:true})
+    }
+
+    public getTitle(txtTitle:string): Cypress.Chainable{
+        return cy.get(this.messageCont).each((item, index) => {
+            expect(item.text().trim()).to.contain(this.message)
+          });
+    }
```
Comment (holgiosalos) [inline cypress/page/upload.page.ts]: why not just simply leave this method to return the content of the div with id `uploaded-files`. So will be just:
`return cy.get(this.messageCont);`

Then, in the test class you made the assertion:
`upload.getTitle().should("have.text", "the file name to verify")`

File: cypress/e2e/upload-download.cy.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+const upload: Upload = new Upload();
+
+describe("Upload and download a file", ()=>{
+  let txt: {targetFile: string};
+  before(()=>{
+    cy.fixture("upload").then((texto)=>{
+      txt = texto;
+    });
+  });
```
Comment (holgiosalos) [inline cypress/e2e/upload-download.cy.ts]: not sure if this is required

File: cypress/e2e/upload-download.cy.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+  it("Should upload a file", ()=>{
+    upload.visitTestPage();
+    upload.uploadFile(txt.targetFile);
```
Comment (holgiosalos) [inline cypress/e2e/upload-download.cy.ts]: I will just pass the filename to upload, by default cypress will search the file in the fixtures folder

File: cypress/e2e/upload-download.cy.ts (hunk 5 above)
Comment (manuelq12) [reply cypress/e2e/upload-download.cy.ts:11]: Well, its a way to resolve a promise and obtain the value so he can use it outside the async flow, i don't mind

File: cypress/e2e/upload-download.cy.ts (hunk 6 above)
Comment (manuelq12) [reply cypress/e2e/upload-download.cy.ts]: Its only required the file name, the method will look for the file in fixtures you don't need anything else.
That means that the file name is testData

**Review by manuelq12 — CHANGES_REQUESTED** [review]

File: cypress/e2e/upload-download.cy.ts (hunk 5 above)
Comment (JuanBohorquezPerficient) [reply cypress/e2e/upload-download.cy.ts]: thanks for the advise, It's already solved

File: cypress/e2e/upload-download.cy.ts (hunk 6 above)
Comment (JuanBohorquezPerficient) [reply cypress/e2e/upload-download.cy.ts]: Great!! I've already solved this one too. Thx, guys

File: cypress/page/upload.page.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        cy.visit(this.testUrl);
+    };
+
+    public uploadFile(targetFile: string): void{
+        cy.get(this.fileContainer).attachFile(targetFile);
+        cy.get(this.submitBtn).click({force:true})
+    };
+
+    public getTitle(txtTitle:string): Cypress.Chainable{
+        return cy.get(this.messageCont).contains(this.message)
```
Comment (holgiosalos) [inline cypress/page/upload.page.ts]: I think that just `return cy.get(this.messageCont);` is enough. The input param is not needed also

File: cypress/e2e/upload-download.cy.ts [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+describe("Upload and download a file", ()=>{
+  const targetFile = "textForTest.txt";
+  before(()=>{
+    cy.fixture("upload");
+  });
+
+  it("Should upload a file", ()=>{
+    upload.visitTestPage();
+    upload.uploadFile(targetFile);
+    upload.getTitle(targetFile);
```
Comment (holgiosalos) [inline cypress/e2e/upload-download.cy.ts]: here add an assertion to verify that title is equal to the targetFile

File: cypress/page/upload.page.ts [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    };
+};
+
+
+
```
Comment (holgiosalos) [inline cypress/page/upload.page.ts]: leave just one blank line

Comment (JuanBohorquezPerficient) [reply cypress/page/upload.page.ts]: ok, sorry... fixed

**Review by holgiosalos — APPROVED** [review]

**Pull request merged**